flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -510,4 +509,39 @@ private int generatePriority(ResourceProfile resourceProfile) {
 		}
 	}
 
+	/**
+	 * Implements NMClientAsyncCallbackHandler.
+	 */
+	private class NMClientAsyncCallbackHandler implements NMClientAsync.CallbackHandler {
```

Comment:
Maybe we could let YarnResourceManager to implement this interface, as it has implemented the `AMRMClientAsync.CallbackHandler`.

Reply:
Yea. Will change.